File: algorithm/61. Rotate List/61.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if(!head || k==0) return head;
        ListNode* fast,*slow;
        int len=1;
        fast=slow=head;
        while(fast->next)
        {
            len++;
            fast=fast->next;
        }
        fast->next=head;
        if(k%=len)
        {
            for(int i=0; i<len-k;i++)
                fast=fast->next;
        }
        slow=fast->next;
        fast->next=NULL;
        return slow;
    }
};

File: algorithm/61. Rotate List/61.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def rotateRight(self, head, k):
        """
        :type head: ListNode
        :type k: int
        :rtype: ListNode
        """
        if(not head):
            return head
        fast=slow=head
        len=1
        while(fast.next):
            fast=fast.next
            len+=1
        fast.next=head
        k%=len
        if k>0:
            for i in range(len-k):
                fast=fast.next
        slow=fast.next
        fast.next=None
        return slow
